common-rendering/src/components/MetaData.tsx

```diff
@@ -0,0 +1,363 @@
+}; */
+
+/*export const bylineAsTokens = (byline: string, tags: TagType[]): string[] => {
+	const titles = getContributorTags(tags).map((c) => c.title);
+	// The contributor tag title should exist inside the byline for this regex to work
+
+	const regex = new RegExp(
+		`(${applyCleverOrderingForMatching(titles).join("|")})`
+	);
+
+	return byline.split(regex);
+}; */
+
+/*export const getContributorTagsForToken = (
+	tags: TagType[],
+	token: string
+): TagType[] => {
+	return getContributorTags(tags).filter((t) => t.title === token);
+};*/
+
+/* export const getBylineComponentsFromTokens = (
+	tokens: string[],
+	tags: TagType[]
+): BylineComponent[] => {
+	const [bylineComponents] = tokens.reduce(
+		([bylines, remainingTags], token) => {
+			const [firstContributorTag] = getContributorTagsForToken(
+				remainingTags,
+				token
+			);
+			if (!firstContributorTag) {
+				return [bylines.concat(token), remainingTags];
+			}
+
+			// We've used this tag, so remove it from the list of possible tags.
+			// This ensures we don't use the same contributor tag when two identical
+			// names with different contributor tags are used.
+			const newRemainingTags = remainingTags.filter(
+				(tag) => !(tag.id === firstContributorTag.id)
+			);
+
+			return [
+				bylines.concat({ tag: firstContributorTag, token }),
+				newRemainingTags,
+			];
+		},
+		[[], tags] as [BylineComponent[], TagType[]]
+	);
+
+	return bylineComponents;
+}; */
+
+/* const BylineLink = ({ byline, tags }: Props) => {
+	const tokens = bylineAsTokens(byline, tags);
+
+	const bylineComponents = getBylineComponentsFromTokens(tokens, tags);
+
+	const renderedTokens = bylineComponents.map((bylineComponent) => {
+		if (typeof bylineComponent === "string") {
+			return bylineComponent;
+		}
+		return (
+			<ContributorLink
+				contributor={bylineComponent.token}
+				contributorTagId={bylineComponent.tag.id}
+				key={bylineComponent.tag.id}
+			/>
+		);
+	});
+
+	return <>{renderedTokens}</>;
+}; */
+
+/*
+	Foreach author in authors
+	if string contains name replace with authorlink then return <p> byline with links
+*/
+
+/* export const getContributorTags = (tags: TagType[]): TagType[] => {
+	return tags.filter((t) => t.type === "Contributor");
+}; */
+
+/* const getAuthorName = (tags: TagType[]) => {
+	const contributorTag = tags.find((tag) => tag.type === "Contributor");
+	return contributorTag && contributorTag.title;
+}; *
+
+type AuthorLinkDetails = {
+	id: string;
+	name: string;
+};
+
+const getAuthorNames = (tags: TagType[]) => {
+	const names: AuthorLinkDetails[] = [];
+	tags.find((tag) => {
+		if (tag.type === "Contributor") {
+			names.push({ name: tag.title, id: tag.id });
+		}
+	});
+};
+
+/*function Byline(byline: string, authors: Authors[]) {
+	if (byline.includes("now")) {
+		byline.replace("now", "(now)");
+	}
+	if (byline.includes("earlier")) {
+		byline.replace("earlier", "(earlier)");
+	}
+}*/
+
+const ContributorLink: React.FC<{
+	contributor: string;
```

Comment:
Could we extract these types (`contributor` & `contributorTag`) to a `ContributorLinkProps` definition or similar?

Reply:
Out of curiosity what would be the benefit of that? I copied this from DCR as is.

Reply:
Out of curiosity what would be the benefit for such a small component? It was re-used from DCR as is.

Reply:
Just convention really - it's how we've always done it on AR. I think beyond that it just makes it slightly easier to read: the component is split over less lines but tbh it's pretty minor.

Reply:
Done